### ISSUE TITLE
'glib.h' file not found.

### ISSUE THREAD

**Issue by sgxadhith** [Homebrew/homebrew-core#42930]
- [x] are reporting a bug others will be able to reproduce and not asking a question or requesting software. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh. To get software added or changed in Homebrew please file a [Pull Request](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)
- [x] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, Homebrew/homebrew-core formula (not cask) on macOS? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a Linux problem please file this issue at https://github.com/Homebrew/linuxbrew-core/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [ ] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [ ] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

I ran the below commands.
```
$ brew update
$ brew install glib
```
I received  the below output.
```
==> Pouring glib-2.60.6.mojave.bottle.tar.gz
🍺  /usr/local/Cellar/glib/2.60.6: 429 files, 15.2MB
```

I take it that this means that glib was installed on my machine. Here is my ```$PATH```.
```
/Users/sanjayadhith/.opam/default/bin:/usr/local/opt/qt/bin:/usr/local/opt/ruby/bin:/Users/sanjayadhith/.cargo/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/share/dotnet:/Library/Frameworks/Mono.framework/Versions/Current/Commands
```

However, when I tried to compile the howl text editor I persistently get this error message despite running ```brew doctor``` and ```brew reinstall glib```.

```
Package libffi was not found in the pkg-config search path.
Perhaps you should add the directory containing `libffi.pc'
to the PKG_CONFIG_PATH environment variable
Package 'libffi', required by 'gobject-2.0', not found
cc -c main.c -Wall -O2 -g -I/Users/sanjayadhith/Downloads/howl-master/src/deps/LuaJIT-2.1.0-beta3/src  -DHOWL_PREFIX=/usr/local
In file included from main.c:4:
./main.h:9:10: fatal error: 'glib.h' file not found
#include <glib.h>
         ^~~~~~~~
1 error generated.
make: *** [main.o] Error 1
```

My installed packages are listed below.
```
apr				jemalloc			ocaml
apr-util			jpeg				opam
atk				leiningen			openexr
cairo				libcaca				openjpeg
cask				libde265			openssl
chicken				libepoxy			openssl@1.1
cmake				libevent			p11-kit
cmatrix				libffi				pandoc
coffeescript			libheif				pango
coreutils			libidn2				pcre
cowsay				libmpc				pcre2
crystal				libomp				perl
curl				libpng				pixman
cvs				libssh2				pkg-config
cython				libtasn1			python
elixir				libtermkey			python@2
emacs				libtiff				qt
erlang				libtool				readline
fontconfig			libunistring			ripgrep
fortune				libuv				ruby
freetype			libvterm			s-lang
fribidi				libyaml				screenresolution
gcc				little-cms2			shared-mime-info
gdbm				llvm@6				sqlite
gdk-pixbuf			lua				subversion
gettext				lua@5.1				thefuck
git				luajit				tmux
glib				luarocks			toilet
gmp				lz4				tree
gnutls				lzo				typescript
go				md5sha1sum			unbound
graphite2			mercurial			unibilium
groovy				midnight-commander		utf8proc
gsettings-desktop-schemas	mit-scheme			vim
gtk+3				mosml				webp
harfbuzz			mpfr				wget
hicolor-icon-theme		msgpack				wine
htop				nasm				wtf
icu4c				ncurses				wxmac
idris				neofetch			x265
ilmbase				neovim				xz
imagemagick			nettle				youtube-dl
isl				node				zsh
```

As this is my first issue on GitHub, I may be missing out on a few details. Please notify me if that is the case. Thank you very much.

**Comment by sgxadhith** [comment]
Checklist added to the issue!

**Comment by refi64** [comment]
I believe the main point of interest is this:

```
Package libffi was not found in the pkg-config search path.
Perhaps you should add the directory containing `libffi.pc'
to the PKG_CONFIG_PATH environment variable
Package 'libffi', required by 'gobject-2.0', not found
```

**Comment by Amar1729** [comment]
Looks like you should do:
`export PKG_CONFIG_PATH="/usr/local/opt/libffi/lib/pkgconfig"`

but then you run into 
```
==== Building LuaJIT 2.1.0-beta3 ====
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C src
HOSTLINK  host/minilua
ld: library not found for -lgcc_s.10.4
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [host/minilua] Error 1
make[1]: *** [default] Error 2
make: *** [deps/LuaJIT-2.1.0-beta3/src/libluajit.a] Error 2
make: *** Waiting for unfinished jobs....
```

Found this "solution" which doesn't seem safe at all:
`cd /usr/local/lib sudo ln -s ../../lib/libSystem.B.dylib libgcc_s.10.4.dylib`

**Comment by fxcoudert** [comment]
1. `pkg-config` files for `glib` work fine:
```
$ pkg-config --libs glib-2.0    
-L/usr/local/Cellar/glib/2.62.0/lib -L/usr/local/opt/gettext/lib -lglib-2.0 -lintl
$ pkg-config --cflags glib-2.0 
-I/usr/local/Cellar/glib/2.62.0/include/glib-2.0 -I/usr/local/Cellar/glib/2.62.0/lib/glib-2.0/include -I/usr/local/opt/gettext/include -I/usr/local/Cellar/pcre/8.43/include
```

2. If you compile software that explicitly needs `pkg-config` to find `libffi`, you will need to set `PKG_CONFIG_PATH` to `/usr/local/opt/libffi/lib/pkgconfig`, indeed. This is the correct fix for your issue.

3. This error:

```
HOSTLINK  host/minilua
ld: library not found for -lgcc_s.10.4
```

is a luajit issue, which can be fixed by setting `MACOSX_DEPLOYMENT_TARGET`. It is not related to Homebrew.

**Comment by nox** [comment]
@fxcoudert No, `glib` doesn't work correctly with keg-only `libffi`.

```
± pkg-config --cflags gobject-2.0
Package libffi was not found in the pkg-config search path.
Perhaps you should add the directory containing `libffi.pc'
to the PKG_CONFIG_PATH environment variable
Package 'libffi', required by 'gobject-2.0', not found
```

See a fix for that in #44446.
